keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/proposal/GaussianProposalDistribution.java

```diff
@@ -0,0 +1,41 @@
+package io.improbable.keanu.algorithms.mcmc.proposal;
+
+import java.util.Set;
+
+import io.improbable.keanu.distributions.ContinuousDistribution;
+import io.improbable.keanu.distributions.continuous.Gaussian;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Probabilistic;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+public class GaussianProposalDistribution implements ProposalDistribution {
+
+    private final DoubleTensor sigma;
+
+    public GaussianProposalDistribution(DoubleTensor sigma) {
+        this.sigma = sigma;
+    }
+
+    @Override
+    public Proposal getProposal(Set<Vertex> vertices, KeanuRandom random) {
+        Proposal proposal = new Proposal();
+        for (Vertex vertex : vertices) {
+            ContinuousDistribution proposalDistribution = Gaussian.withParameters((DoubleTensor) vertex.getValue(), sigma);
```

Comment:
Casts are icky.  I wonder if we could fix up the class hierarchy somewhat to remove the need for this?

Reply:
Agreed, casts are icky. The problem here (and many other places) is that we need a collection of vertices of mixed types. 

In fact I wonder if I ought to be filtering out those that are not `DoubleVertex` here. @gordoncaleb - can a proposal contain non-double-values vertices?